### ISSUE TITLE
[Feature] Unittest for using a browser extension with `launch_persistent_context`

### ISSUE THREAD

**Issue by palewire** [microsoft/playwright#14461]
[The Playwright docs](https://playwright.dev/python/docs/chrome-extensions) encourage the use of `launch_persistent_context` in headful mode in order to include a browser extension. I have been attempting to do this and struggling hard. While the code works flawlessly on my laptop, GitHub Actions using xvfb are failing. 

I've filed a ticket in the downstream Python library [here](https://github.com/microsoft/playwright-python/issues/1277). Logs of the error can be found [here](https://github.com/palewire/news-homepages/runs/6637493508?check_suite_focus=true). The relevant workflow file is [here](https://github.com/palewire/news-homepages/blob/ad-blocking/.github/actions/screenshot/action.yml). The Python code that launches the browser is [here](https://github.com/palewire/news-homepages/blob/ad-blocking/newshomepages/screenshot.py).

Here's my request: This core library should add a unittest that verifies that the recommended use of `launch_persistent_context` with a browser extension. This would be similar to [the existing test that captures video](https://github.com/microsoft/playwright/blob/main/.github/workflows/tests_video.yml).

**Comment by yury-s** [comment]
[Your script](https://github.com/palewire/news-homepages/blob/b6f30712e29cdb91abaca40ae02a24102211b8f2/.github/actions/screenshot/action.yml#L41-L43) that launches xvfb looks suspecios, try replacing it with something like this: `xvfb-run --auto-servernum --server-args="-screen 0 1280x960x24" -- <your command>`.

Please don't file duplicate issues for the same problem.

**Comment by palewire** [comment]
Thanks for the tip. I made the change you recommended and got the same result. Seen [here](https://github.com/palewire/news-homepages/runs/6645571763?check_suite_focus=true)

**Comment by aslushnikov** [comment]
> Here's my request: This core library should add a unittest that verifies that the recommended use of launch_persistent_context with a browser extension. 

@palewire Here's one:

https://github.com/microsoft/playwright/blob/da55afa5847d0d24e3a7350ea8bcf3a8bf2d1590/tests/library/chromium/launcher.spec.ts#L55-L76

Overall, it looks like things don't work for you with XVFB and GitHub actions. We have coverage for this in our repos, so I'm inclined to believe that the issue is on your side. You can prove me wrong though :)

To debug, can you start with very basic things and launch headed chrome on github actions, without anything else? Something like this (I'll type in JS):

```js
import { chromium } from 'playwright';

const browser = await chromium.launch({ headless: false });
const page = await browser.newPage();
// this should not have a "headless" word!
console.log(await page.evaluate(() => navigator.userAgent));
await browser.close();
```

**Comment by palewire** [comment]
Thank you for point out this unittest, @aslushnikov. I was unable to locate it when I filed the ticket. Its code, combined with your advice, has given me a route forward. Thanks for that too.

I added some logging steps, as you recommended. When I substituted in the "simple-extension" from the unittests, my headful script did work. While not an ultimate solution it does narrow down my bug hunt a bit. My new provisional guess is that the uBlock extension I'm trying to implement has some issue inside of GitHub Actions. Since it works on my local laptop, I suspect it must be something about how it is executed inside of the cloud server. But I'm still just guessing.

Since my initial idea for a new feature has already been implemented, I'll close this ticket. If anyone has any further advice about getting uBlock to work with Playwright in Actions I'd appreciate the tip. My hunch is that this would be a very useful use case for others besides me.
